envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -18,7 +18,15 @@ option (gogoproto.equal_all) = true;
 
 // Upstream host identifier.
 message Endpoint {
+  // The upstream host address.
   core.Address address = 1;
+
+  // [#not-implemented-hide:] Optional alternative health check port.
```

Comment:
How general is this? I'm wondering what other parameters might be configurable at the per-endpoint level, e.g. might we want to health check using TLS and a different cert, etc? Not really advocating to make this more complicated, but would like to make sure we have covered the potential endpoint health check specializations at least at the design level.

Reply:
Probably [this thread](https://envoyproxy.slack.com/archives/C78M4KW76/p1522377037000064) on #envoy-users could be related too?

Reply:
+1 at least from the proto perspective. Can you maybe add a new message such as `HealthCheckConfig` (or something like that), and then put the port override in there? Then we can add other things in the future if needed.

From a doc perspective, we should make it clear that health checking must be configured at the cluster level for these options to take any effect, etc.